src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -482,50 +481,46 @@ public Script getScriptSigWithSignature(Script scriptSig, byte[] sigBytes, int i
         return ScriptBuilder.updateScriptWithSignature(scriptSig, sigBytes, index, sigsPrefixCount, sigsSuffixCount);
     }
 
-    private RedeemScriptParser getRedeemScriptParser() {
-        if (redeemScriptParser == null){
-            redeemScriptParser = RedeemScriptParserFactory.get(chunks);
-        }
-        return redeemScriptParser;
-    }
-
     /**
      * Returns the index where a signature by the key should be inserted. Only applicable to
      * a P2SH scriptSig.
      */
     public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
+        List<ScriptChunk> rawChunks = this.getChunks();
         // Iterate over existing signatures, skipping the initial OP_0, the final redeem script
         // and any placeholder OP_0 sigs.
-        List<ScriptChunk> existingChunks = chunks.subList(1, chunks.size() - 1);
-        ScriptChunk redeemScriptChunk = chunks.get(chunks.size() - 1);
-        checkNotNull(redeemScriptChunk.data);
-        Script redeemScript = new Script(redeemScriptChunk.data);
+        List<ScriptChunk> existingChunks = rawChunks.subList(1, rawChunks.size() - 1);
+        RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(this.getChunks());
 
         int sigCount = 0;
-        int myIndex = redeemScript.findKeyInRedeem(signingKey);
-        for (ScriptChunk chunk : existingChunks) {
-            if (chunk.opcode != OP_0) {
-                checkNotNull(chunk.data);
-                if (myIndex < redeemScript.findSigInRedeem(chunk.data, hash)) {
+        int myIndex = redeemScriptParser.findKeyInRedeem(signingKey);
+        Iterator chunkIterator = existingChunks.iterator();
+
+        while(chunkIterator.hasNext()) {
+            ScriptChunk chunk = (ScriptChunk) chunkIterator.next();
+            if (chunk.opcode != 0) {
```

Comment:
Is this the same as comparing against `OP_0`?

Reply:
It is but I also updated this. The new version is using `OP_0`.